interface/product_registration/product_registration_service.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * ProductRegistrationService
+ *
+ * LICENSE: This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://opensource.org/licenses/gpl-license.php>;.
+ *
+ * @package OpenEMR
+ * @author  Matthew Vita <matthewvita48@gmail.com>
+ * @link    http://www.open-emr.org
+ */
+
+$fake_register_globals=false;
+$sanitize_all_escapes=true;
+include_once("../globals.php");
+require_once($GLOBALS['fileroot'] . "/interface/main/exceptions/invalid_email_exception.php");
+require_once($GLOBALS['fileroot'] . "/interface/product_registration/local_product_registration_exception.php");
+require_once($GLOBALS['fileroot'] . "/interface/product_registration/remote_product_registration_exception.php");
+require_once($GLOBALS['fileroot'] . "/ccr/uuid.php");
+
+// TODO: @Brady to help me get this in the migration DSL...
+// TODO: Figure out why foreign key returns `errno 150`... am thinking it's because user email isn't uniq.
+//CREATE TABLE product_registration (
+//    registration_id CHAR(36) NULL,
+//    email VARCHAR(255) NULL,
+//    opt_out TINYINT(1) NULL,
+//    PRIMARY KEY (registration_id)/*
+//    FOREIGN KEY (email) REFERENCES users(email)*/
```

Comment:
This will be the first foreign key in openemr. Also ensure to force the InnoDB engine.

Table first goes into(which is used on OpenEMR install):
sql/database.sql

and when it's solid, then will copy it into sql/4_2_2-to-5_0_0_upgrade.sql (for upgrading) inside a bit of crazy code:
``````
#IfNotTable
<table code>
#EndIf